core/src/components/datetime/datetime.tsx

```diff
@@ -1176,7 +1189,7 @@ export class Datetime implements ComponentInterface {
 
     const { minParts, maxParts, multiple } = this;
     if (!multiple && Array.isArray(value)) {
-      this.value = value[0];
+      this.setValue(value[0]);
```

Comment:
`processValue` only gets called when the datetime has a default value set (or when resetting to said default via `reset()`) so I don't think we'd want to fire `ionChange` here?

If `ionChange` should fire when resetting, we could manually fire it in `reset`, though we'd also need to remove the `ionChange` fire in `clearButtonClick` below since that calls `reset`.

Reply:
Not sure I follow. In the line you highlighted, the `value` would change from something like`['2022-01-01', '2022-01-02']` to `'2022-01-01'` which is the same process that the accordion group follows: https://github.com/ionic-team/ionic-framework/blob/b052d3b2622b795cde102591f0191338e15b14a0/core/src/components/accordion-group/accordion-group.tsx#L86

Reply:
The difference is that this line is from a function that's only called when initing the datetime with a defined default value, or resetting to that default value. So if you had something like this (Vue syntax):
```
<ion-datetime :multiple="false" :value="['2022-01-01', '2022-01-02', '2022-01-03']"></ion-datetime>
```
... that would cause `ionChange` to fire immediately on load when the `value` is updated from an array to a single value (due to `multiple` being `false`). So, I believe we should just set `this.value` directly instead of going through `setValue`, to avoid the `ionChange` fire. I'm cool with keeping it as-is if that's desired behavior; just checking since it struck me as a little weird.

Reply:
Discussed offline. We are going to change datetime to no longer adjust the value when used incorrectly. Instead we will warn/error in the console.